src/scrapers/mizrahi.js

```diff
@@ -0,0 +1,155 @@
+    `${AFTER_LOGIN_BASE_URL}/main/uis/ge/changePassword/`,
+  ];
+  return urls;
+}
+
+function CreateDataFromRequest(request, optionsStartDate) {
+  const defaultStartMoment = moment().subtract(1, 'years');
+  const startDate = optionsStartDate || defaultStartMoment.toDate();
+  const startMoment = moment.max(defaultStartMoment, moment(startDate));
+
+  const data = JSON.parse(request.postData());
+
+  data.inToDate = moment().format(DATE_FORMAT);
+  data.inFromDate = startMoment.format(DATE_FORMAT);
+  data.table.maxRow = MAX_ROWS_PER_REQUEST;
+
+  return data;
+}
+
+function createHeadersFromRequest(request) {
+  return {
+    mizrahixsrftoken: request.headers().mizrahixsrftoken,
+    'Content-Type': request.headers()['content-type'],
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((row) => {
+    // row.MC02PeulaTaaEZ: 2019-09-10T00:00:00
+    const txnDate = moment(row.MC02PeulaTaaEZ, moment.HTML5_FMT.DATETIME_LOCAL_SECONDS)
+      .format(ISO_DATE_FORMAT);
+
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: row.MC02AsmahtaMekoritEZ ? parseInt(row.MC02AsmahtaMekoritEZ, 10) : null,
+      date: txnDate,
+      processedDate: txnDate,
+      originalAmount: row.MC02SchumEZ,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: row.MC02SchumEZ,
+      description: row.MC02TnuaTeurEZ,
+      status: TRANSACTION_STATUS.COMPLETED,
+    };
+  });
+}
+
+async function extractPendingTransactions(page) {
+  const pendingTxn = await pageEvalAll(page, 'tr.rgRow', [], (trs) => {
+    return trs.map((tr) => Array.from(tr.querySelectorAll('td'), (td) => td.textContent));
+  });
+
+  return pendingTxn.map((txn) => {
+    const date = moment(txn[0], 'DD/MM/YY').format(ISO_DATE_FORMAT);
+    const amount = parseInt(txn[3], 10);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: null,
+      date,
+      processedDate: date,
+      originalAmount: amount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: amount,
+      description: txn[1],
+      status: TRANSACTION_STATUS.PENDING,
+    };
+  });
+}
+
+class MizrahiScraper extends BaseScraperWithBrowser {
+  async sniffOshRequest() {
+    await waitForNavigation(this.page);
+    this.navigateTo(OSH_PAGE, this.page);
+    this.request = await this.page.waitForRequest(TRANSACTIONS_REQUEST_URL);
+  }
+
+  getLoginOptions(credentials) {
+    return {
+      loginUrl: `${LOGIN_URL}`,
+      fields: createLoginFields(credentials),
+      submitButtonSelector: '#ctl00_PlaceHolderLogin_ctl00_Enter',
+      postAction: async () => this.sniffOshRequest(),
+      possibleResults: getPossibleLoginResults(),
+    };
+  }
+
+  async fetchData() {
+    const data = CreateDataFromRequest(this.request, this.options.startDate);
+    const headers = createHeadersFromRequest(this.request);
+
+    const response = await fetchPostWithinPage(this.page,
+      TRANSACTIONS_REQUEST_URL, data, headers);
+
+    if (response.header.success === false) {
```

Comment:
@baruchiro we prefer to have `if (!response.header.success)`

Reply:
Sorry, my fix is broken.
If something is wrong in our request, the response will be `response.header.success === false`.
But if everything is OK and the response contain the transaction, **the response will be `response.header.success === null`**.

Reply:
@baruchiro 😞 that's terrible. these banks and their awful API definitions. fine, let's keep it.